### ISSUE TITLE
skadi (valhalla_build_elevation) and compressed hgt tiles

### ISSUE THREAD

**Issue by molind** [valhalla/valhalla#879]
Just suggestion of optimization.

I saw you compressed hgt files using gz on your server. And that's perfect option to save bandwidth. But may be it worth also to support compressed hgt files inside skadi to save some storage. Typical compression ratio is 10-20%. It's 5-10 times less storage used for elevation data. With small overhead to decompress on the fly. As you may know there is archive formats with very fast decompression speed.

**Comment by Colleoni** [comment]
Hi
Is there any news about this topic? I saw that valhalla_build_elevation still uncompress the hgt.gz files.
However, it seems to me that Skadi does handle .gz.

Am I correct? Does skadi support .gz files now?
Thanks

**Comment by kevinkreiser** [comment]
That is correct, skadi supports gzipped files however its quite a bit slower than uncompressed files. It might work for your use case if you need smaller data vs service speed something like 300gb instead of 1.6tb but yeah its quite a bit slower and the caching mechanism is pretty dumb. `valhalla_build_elevation` will optionally not decompress the files but it defaults to doing so: https://github.com/valhalla/valhalla/blob/master/scripts/valhalla_build_elevation#L5

**Comment by kevinkreiser** [comment]
I'll also mention i did implement on the fly decompression via lz4 but the dependency for that was quite a bit more annoying than zlib so eventually we removed support for this.

**Comment by kevinkreiser** [comment]
this was completed quite a while ago! it could be further enhanced please open another issue if we need to do some more work here!

**Comment by Colleoni** [comment]
Thank you for your response, 
I spent the day trying the compressed format. After I wrote in this issues, I saw lz4 was introduced in 2.4.4 but I did not know it has been also removed, I suggest to mention this in the changelog.
I tried hgt.gz but I think my version of valhalla is too old, it is not even mentioned in the changelog:) I will try it again with a newer version. I think I will continue trying hg.gz as I use only the elevation API and not really heavily and reaching 300 GB from 1.6TB is quite an improvement.
Thanks again for the support
